app/forms/idv/address_form.rb

```diff
@@ -13,46 +13,41 @@ def self.model_name
       ActiveModel::Name.new(self, nil, 'IdvForm')
     end
 
-    def initialize(pii)
-      set_ivars_with_pii(pii)
-      @address_edited = false
+    def initialize(initial_address)
+      consume_attributes_from_address(initial_address.to_h)
     end
 
     def submit(params)
-      consume_params(params)
+      consume_attributes_from_address(params.to_h)
 
       FormResponse.new(
         success: valid?,
         errors: errors,
         extra: {
-          address_edited: @address_edited,
           pii_like_keypaths: [[:errors, :zipcode], [:error_details, :zipcode]],
         },
       )
     end
 
-    private
-
-    def set_ivars_with_pii(pii)
-      pii = pii.symbolize_keys
-      @address1 = pii[:address1]
-      @address2 = pii[:address2]
-      @city = pii[:city]
-      @state = pii[:state]
-      @zipcode = pii[:zipcode]
+    def updated_user_address
+      Pii::Address.new(
+        address1: address1,
+        address2: address2,
+        city: city,
+        state: state,
+        zipcode: zipcode,
+      )
     end
 
-    def consume_params(params)
-      ATTRIBUTES.each do |attribute_name|
-        if send(attribute_name).to_s != params[attribute_name].to_s
-          @address_edited = true
-        end
-        send(:"#{attribute_name}=", params[attribute_name].to_s)
-      end
-    end
+    private
 
-    def raise_invalid_address_parameter_error(key)
-      raise ArgumentError, "#{key} is an invalid address attribute"
+    def consume_attributes_from_address(address_hash)
+      address_hash = address_hash.symbolize_keys
+      @address1 = address_hash[:address1]
+      @address2 = address_hash[:address2]
+      @city = address_hash[:city]
+      @state = address_hash[:state]
+      @zipcode = address_hash[:zipcode]
     end
```

Comment:
would it make sense to call `.strip` or any other normalization on these here?

Reply:
Perhaps, but it might be redundant. Currently the app just calls `#to_s` on whatever comes through the params.